client.js

```diff
@@ -20,24 +20,25 @@ module.exports = class Client extends EventEmitter {
    * @constructor
    * @param {Object} obj - The properties of the client.
    * @param {String} [obj.name] - The client's name, used for debugging messages.
+   * @param {String} [obj.password] - The client's password, used for generating address.
    * @param {Object} obj.net - The network used by the client
    *    to send messages to all miners and clients.
    * @param {Block} [obj.startingBlock] - The starting point of the blockchain for the client.
    * @param {Object} [obj.keyPair] - The public private keypair for the client.
    */
-  constructor({name, net, startingBlock, keyPair} = {}) {
+  constructor({name, password, net, startingBlock, keyPair} = {}) {
     super();
 
     this.net = net;
     this.name = name;
 
-    if (keyPair === undefined) {
-      this.keyPair = utils.generateKeypair();
-    } else {
-      this.keyPair = keyPair;
-    }
+    this.password = password ? password : this.name+"_pswd";
 
-    this.address = utils.calcAddress(this.keyPair.public);
+
+    if (Blockchain.isInstance()){
```

Comment:
Can you test out what happens when the client is not initialized yet?  (I want to make sure that we don't introduce any additional bugs).

Reply:
mnemonic-driver.js: have added a late client and included a transaction involving the late client